world_time_app/lib/services/world_time.dart

```diff
@@ -2,40 +2,47 @@ import 'package:http/http.dart';
 import 'dart:convert';
 import 'package:intl/intl.dart';
 
-class WorldTime {
+class WorldTime{
+  String location; // location name for the UI
+  String time; // time in that location
+  String flag; // url to asset flag icon
+  String url; // location url for API endpoint
+  bool isDayTime; // true or false / day or night
 
-  String location; // location name for UI
-  String time; // the time in that location
-  String flag; // url to an asset flag icon
-  String url; // location url for api endpoint
-  bool isDaytime; // true or false if daytime or not
-
-  WorldTime({ this.location, this.flag, this.url });
+  WorldTime({ this.location, this.flag, this.url});
 
   Future<void> getTime() async {
 
-    try{
-      // make the request
+    try {
+      //make the request
       Response response = await get('http://worldtimeapi.org/api/timezone/$url');
       Map data = jsonDecode(response.body);
+      print(data);
+
 
-      // get properties from json
+      //get properties from data
       String datetime = data['datetime'];
-      String offset = data['utc_offset'].substring(1,3);
+      int offsetHours = int.parse(data['utc_offset'].substring(1,3));
+      int offsetMinutes = int.parse(data['utc_offset'].substring(4,6));
+
+      if (data['utc_offset'].substring(0, 1) == "-") {
+        offsetHours = -offsetHours;
+        offsetMinutes = -offsetMinutes;
```

Comment:
`now  = now.subtract(Duration(hours: offsetHours, minutes: offsetMinutes));
No need to change offset sign